redis/src/commands/mod.rs

```diff
@@ -1257,6 +1263,157 @@ implement_commands! {
+    /// [Redis Docs](https://redis.io/commands/VADD)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vadd_extended<K: ToRedisArgs, E: ToRedisArgs>(key: K, input: VectorAddInput<'a>, element: E, options: &'a VAddOptions) -> (bool) {
+        cmd("VADD").arg(key).arg(options.reduction_dimension.map(|_| "REDUCE")).arg(options.reduction_dimension).arg(input).arg(element).arg(options)
+    }
+
+    /// Get the number of members in a vector set.
+    /// [Redis Docs](https://redis.io/commands/VCARD)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vcard<K: ToRedisArgs>(key: K) -> (usize) {
+        cmd("VCARD").arg(key)
+    }
+
+    /// Return the number of dimensions of the vectors in the specified vector set.
+    /// [Redis Docs](https://redis.io/commands/VDIM)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vdim<K: ToRedisArgs>(key: K) -> (usize) {
+        cmd("VDIM").arg(key)
+    }
+
+    /// Return the approximate vector associated with a given element in the vector set.
+    /// [Redis Docs](https://redis.io/commands/VEMB)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vemb<K: ToRedisArgs, E: ToRedisArgs>(key: K, element: E) -> Generic {
+        cmd("VEMB").arg(key).arg(element)
+    }
+
+    /// Return the raw internal representation of the approximate vector associated with a given element in the vector set.
+    /// Vector sets normalize and may quantize vectors on insertion.
+    /// VEMB reverses this process to approximate the original vector by de-normalizing and de-quantizing it.
+    /// [Redis Docs](https://redis.io/commands/VEMB)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vemb_raw<K: ToRedisArgs, E: ToRedisArgs>(key: K, element: E) -> Generic {
+        cmd("VEMB").arg(key).arg(element).arg("RAW")
+    }
+
+    /// Remove an element from a vector set.
+    /// [Redis Docs](https://redis.io/commands/VREM)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vrem<K: ToRedisArgs, E: ToRedisArgs>(key: K, element: E) -> (bool) {
+        cmd("VREM").arg(key).arg(element)
+    }
+
+    /// Associate a JSON object with an element in a vector set.
+    /// Use this command to store attributes that can be used in filtered similarity searches with VSIM.
+    /// [Redis Docs](https://redis.io/commands/VSETATTR)
+    #[cfg(feature = "vector-sets")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "vector-sets")))]
+    fn vsetattr<K: ToRedisArgs, E: ToRedisArgs, J: Serialize>(key: K, element: E, json_object: &'a J) -> (bool) {
+        let attributes_json = match serde_json::to_value(json_object) {
+            Ok(serde_json::Value::String(s)) if s.is_empty() => "".to_string(),
```

Comment:
what happens on the `Err` case? why not fail then?
Also, why bother with the `if s.is_empty()` check? what happens if it is removed?

Reply:
Without this check empty strings will be serialized as `\"\"`, which is not the intended behavior. An empty string should remove the attributes. Without it an empty JSON object will be set as an attribute.

If the matching fails completely `unwrap()` will panic at runtime. The fail case when the passed string is not empty ("") simply tries to convert the passed in object to string.